### PR TITLE
Add single post learning assessment front end

### PR DESCRIPTION
**Pull request by benjimarshall**
## Changes
* Amend section action to redirect to the post learning assessment page if the course only has a post learning assessment.
* Add completion data to post learning assessment view model, to be shown if it is the item in the course, and flags to record whether this is the only section, and only item in the section.
* Hide the next button if this is the only item in the section.
* Modify the mobile-only back breadcrumb to have a more helpful name,
depending on if there are other sections in the course or not.

## Testing
Amend tests for this new information, and new learning menu controller behaviour. Tested in Chrome and IE11 using a screen reader and no JS.

### Single post learning assessment, without a completion summary
single_pl_no_completion

### Single post learning assessment using a narrow viewport, without a completion summary
single_pl_no_completion_narrow

### Single post learning assessment, with a completion summary
single_pl_with_completion

### Section with post learning assessment and consolidation exercise
pl_and_consolidation

### CODE REVIEW

**Comment by benjimarshall** [comment]
The variables names got a bit confused there for a little while, but they should all be sorted now 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs [hunk 1]
```diff
@@ -249,6 +250,210 @@ public void Post_learning_assessment_can_have_no_next_section()
...
+            // When
+            var postLearningAssessmentViewModel =
+                new PostLearningAssessmentViewModel(postLearningAssessment, CustomisationId, SectionId);
+
+            // Then
+            postLearningAssessmentViewModel.OnlyItemInOnlySection.Should().BeTrue();
+        }
+
+        [Test]
+        public void Post_learning_assessment_should_be_only_item_in_this_section_when_no_other_sections_exist()
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs:307]: Where do you specify that no other sections exist? And same for the test below? I'm not sure the test name reflect what you're testing. Maybe a parameterised test that checks expected OnlyItemInThisSection for all possible inputs of otherItemsInSectionExist and otherSectionsExist? And same for OnlyItemInOnlySection? What do you think? 🙂

**Review by FridaTveit — COMMENTED** [review]
One suggestion for the tests, otherwise looks good 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs (hunk 1 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs:307]: This is specified in line 311: `otherItemsInSectionExist: false`, essentially it checks that the value is passed through from the data model to the view model correctly. I'll parameterise these now 🙂

Comment (FridaTveit) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs:307]: But that's specifying that no other items exist in this section right? Whereas the test name says that no other sections exist 🙂

Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs]: Done in 407303e 🙂

Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs]: > But that's specifying that no other items exist in this section right? Whereas the test name says that no other sections exist 🙂

Yeah that test name was wrong, but it was removed when I parameterised the tests 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs [hunk 2]
```diff
@@ -250,163 +250,77 @@ public void Post_learning_assessment_can_have_no_next_section()
...
-            postLearningAssessmentViewModel.OnlyItemInThisSection.Should().BeTrue();
+            postLearningAssessmentViewModel.OnlyItemInThisSection.Should().Be(onlyItemInThisSection);
         }
 
-        [Test]
-        public void Post_learning_assessment_should_not_be_only_item_in_this_section_when_other_sections_exist()
+        [TestCase(false, false, true, true)]
+        [TestCase(false, false, false, false)]
+        [TestCase(true, false, true, false)]
+        [TestCase(false, true, true, false)]
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs:302]: I don't think this covers all cases? I would include all the possible values of the inputs, both to be thorough and to make it read more like a truth table (which I think makes it easier to understand but that might be just me) 🙂

**Review by FridaTveit — COMMENTED** [review]
A couple of comments, otherwise looks good 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs [hunk 3]
```diff
@@ -250,163 +250,77 @@ public void Post_learning_assessment_can_have_no_next_section()
...
+        }
+
+        [TestCase(false, false, true)]
+        [TestCase(false, true, false)]
+        [TestCase(true, false, false)]
+        [TestCase(true, true, false)]
+        public void Post_learning_assessment_should_have_onlyItemInOnlySection(
+            bool otherSectionsExist,
+            bool otherItemsInSectionExist,
+            bool onlyItemInOnlySection
```
[the remark this answers is not in the document]
Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs]: Yeah I agree! Done in ab474fe 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs (hunk 2 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/PostLearningAssessmentViewModelTests.cs:302]: I had just converted the individual tests over, but I agree now that test cases are so concise it is simpler to just add them all. Done in ab474fe 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**